modules/plugins/languages/vala.nix

```diff
@@ -0,0 +1,109 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}: let
+  inherit (builtins) attrNames;
+  inherit (lib.options) mkEnableOption mkOption;
+  inherit (lib.meta) getExe;
+  inherit (lib.nvim.languages) diagnosticsToLua;
+  inherit (lib.modules) mkIf mkMerge;
+  inherit (lib.lists) isList;
+  inherit (lib.types) enum either listOf package str;
+  inherit (lib.nvim.types) mkGrammarOption diagnostics;
+  inherit (lib.nvim.lua) expToLua;
+
+  cfg = config.vim.languages.vala;
+
+  defaultServer = "vala_ls";
+  servers = {
+    vala_ls = {
+      package = pkgs.vala-language-server;
+      lspConfig = ''
```

Comment:
The formatting requires the presence of uncrustify as a dependency, which I couldn't get to work because it required a config file external to nvf (at least, I couldn't get anymore information out of their docs. If anyone knows the project better, do tell)

Reply:
If we won't have formatting, then you passed the wrong function to on_attach

Reply:
Got it working. Will push soon